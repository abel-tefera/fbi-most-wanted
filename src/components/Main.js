import React from 'react';
import fbiLogo from '../assets/fbi-logo.png';

const Main = () => (
  <div className="w-full p-6 flex items-center text-left justify-evenly md:justify-center bg-custom-blue-normal">
    <img alt="FBI Logo" src={fbiLogo} className="w-1/2 w-48" />
    <div className="flex flex-col">
      <span className="text-6xl font-extrabold text-white md:text-8xl">
        FBI
      </span>
      <span className="text-2xl md:text-4xl text-white">Most Wanted</span>
    </div>
  </div>
);

export default Main;
