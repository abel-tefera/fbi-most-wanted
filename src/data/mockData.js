const mockData = {
  aliases: [
    'Abd Al-Rauf Bin Al-Habib Bin Yousef Al-Jiddi',
    'Abderraouf Dey',
    'A. Raouf Jdey',
    "Abdal Ra'Of Bin Muhammed Bin Yousef Al-Jadi",
    'Farouq Al-Tunisi',
    'Abderraouf Ben Habib Jeday',
  ],
  age_max: null,
  age_min: null,
  age_range: null,
  complexion: 'Olive',
  caution: null,
  coordinates: [],
  dates_of_birth_used: ['May 30, 1965'],
  details:
    '<p>Abderraouf Jdey is wanted for questioning in connection with possible terrorist threats against the United States.</p>',
  eyes: 'brown',
  hair: 'brown',
  images: [
    {
      original:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/@@images/image',
      large:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/@@images/image/large',
      caption: 'Passport-Type Photo Found with Suicide Letter',
      thumb:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/@@images/image/thumb',
    },
    {
      original:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey4.jpg',
      large:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey4.jpg/@@images/image/large',
      caption: '',
      thumb:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey4.jpg/@@images/image/thumb',
    },
    {
      original:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey.jpg',
      large:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey.jpg/@@images/image/large',
      caption:
        'FBI Laboratory Photographic Retouch. Jdey is wanted for questioning in connection with possible terrorist threats against the United States.',
      thumb:
        'https://www.fbi.gov/wanted/terrorinfo/abderraouf-jdey/jdey.jpg/@@images/image/thumb',
    },
  ],
  height_max: 72,
  height_min: 72,
  nationality: null,
  occupations: null,
  place_of_birth: 'Tunisia',
  reward_max: 4999999,
  reward_min: 1000000,
  reward_text:
    'The Rewards For Justice Program, United States Department of State, is offering a reward of up to $5 million for information leading directly to the capture of Abderraouf Jdey.',
  sex: 'Male',
  subjects: ['Seeking Information - Terrorism'],
  title: 'ABDERRAOUF JDEY',
  warning_message: 'SHOULD BE CONSIDERED ARMED AND DANGEROUS',
  weight: '209 pounds',
  wanted_id: '8afa8ac1eccbfea19b9d4d9682145b14',
  description: '',
};

export default mockData;
