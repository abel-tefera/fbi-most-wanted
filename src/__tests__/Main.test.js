import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import fetchMock from 'jest-fetch-mock';
import configureStore from 'redux-mock-store';
import mockData from '../data/mockData';
import Main from '../components/Main';

const mockStore = configureStore([]);

describe('tests for main component', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
  });

  it('should fetch data and render them', async () => {
    fetchMock.mockResponseOnce(JSON.stringify(mockData));

    const store = mockStore(mockData);

    const component = render(
      <Provider store={store}>
        <Main />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
