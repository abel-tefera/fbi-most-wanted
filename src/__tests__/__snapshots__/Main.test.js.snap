// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections Component should fetch collections and render them 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full p-6 flex items-center text-left justify-evenly md:justify-center bg-custom-blue-normal"
      >
        <img
          alt="FBI Logo"
          class="w-1/2 w-48"
          src="fbi-logo.png"
        />
        <div
          class="flex flex-col"
        >
          <span
            class="text-6xl font-extrabold text-white md:text-8xl"
          >
            FBI
          </span>
          <span
            class="text-2xl md:text-4xl text-white"
          >
            Most Wanted
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full p-6 flex items-center text-left justify-evenly md:justify-center bg-custom-blue-normal"
    >
      <img
        alt="FBI Logo"
        class="w-1/2 w-48"
        src="fbi-logo.png"
      />
      <div
        class="flex flex-col"
      >
        <span
          class="text-6xl font-extrabold text-white md:text-8xl"
        >
          FBI
        </span>
        <span
          class="text-2xl md:text-4xl text-white"
        >
          Most Wanted
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tests for main component should fetch data and render them 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full p-6 flex items-center text-left justify-evenly md:justify-center bg-custom-blue-normal"
      >
        <img
          alt="FBI Logo"
          class="w-1/2 w-48"
          src="fbi-logo.png"
        />
        <div
          class="flex flex-col"
        >
          <span
            class="text-6xl font-extrabold text-white md:text-8xl"
          >
            FBI
          </span>
          <span
            class="text-2xl md:text-4xl text-white"
          >
            Most Wanted
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full p-6 flex items-center text-left justify-evenly md:justify-center bg-custom-blue-normal"
    >
      <img
        alt="FBI Logo"
        class="w-1/2 w-48"
        src="fbi-logo.png"
      />
      <div
        class="flex flex-col"
      >
        <span
          class="text-6xl font-extrabold text-white md:text-8xl"
        >
          FBI
        </span>
        <span
          class="text-2xl md:text-4xl text-white"
        >
          Most Wanted
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
